### PR TITLE
[Merged by Bors] - feat(Analysis/Convex): lemmas about low-dimensional `stdSimplex`es

### PR DESCRIPTION
**Pull request by urkud**
Forward-port of leanprover-community/mathlib#19101
Motivated by https://github.com/Shamrock-Frost/BrouwerFixedPoint

Co-authored-by: @Shamrock-Frost

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Shamrock-Frost** [comment]
Thank you so much for working on this Yury!

File: Mathlib/Analysis/Convex/Basic.lean [hunk 1]
```diff
@@ -688,11 +690,58 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
     exact hab
 #align convex_std_simplex convex_stdSimplex
 
-variable {ι}
+@[nontriviality] lemma stdSimplex_of_subsingleton [Subsingleton 𝕜] : stdSimplex 𝕜 ι = univ :=
+  eq_univ_of_forall fun _ ↦ ⟨fun _ ↦ (Subsingleton.elim _ _).le, Subsingleton.elim _ _⟩
+
+/-- The standard simplex in the zero-dimensional space is empty. -/
+lemma stdSimplex_of_isEmpty [IsEmpty ι] [Nontrivial 𝕜] : stdSimplex 𝕜 ι = ∅ :=
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Basic.lean:697]: Maybe
```suggestion
lemma stdSimplex_of_isEmpty_index [IsEmpty ι] [Nontrivial 𝕜] : stdSimplex 𝕜 ι = ∅ :=
```
Or something to distinguish it from the lemma above

File: Mathlib/Analysis/Convex/Basic.lean [hunk 2]
```diff
@@ -688,11 +690,58 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
...
 
-theorem ite_eq_mem_stdSimplex (i : ι) : (fun j => ite (i = j) (1 : 𝕜) 0) ∈ stdSimplex 𝕜 ι :=
-  ⟨fun j => by simp only; split_ifs <;> norm_num, by
-    rw [Finset.sum_ite_eq, if_pos (Finset.mem_univ _)]⟩
+variable {ι} [DecidableEq ι]
+
+theorem single_mem_stdSimplex (i : ι) : Pi.single i 1 ∈ stdSimplex 𝕜 ι :=
+  ⟨le_update_iff.2 ⟨zero_le_one, fun _ _ ↦ le_rfl⟩, by simp⟩
+
+theorem ite_eq_mem_stdSimplex (i : ι) : (fun j => ite (i = j) (1 : 𝕜) 0) ∈ stdSimplex 𝕜 ι := by
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Basic.lean:711]:
```suggestion
theorem ite_eq_mem_stdSimplex (i : ι) : (fun j => if i = j then (1 : 𝕜) else 0) ∈ stdSimplex 𝕜 ι := by
```

File: Mathlib/Analysis/Convex/Basic.lean [hunk 3]
```diff
@@ -688,11 +690,58 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
...
+  simp
+
+end OrderedSemiring
+
+section OrderedRing
+
+variable (𝕜) [OrderedRing 𝕜]
+
+/-- The standard one-dimensional simplex in `Fin 2 → 𝕜` is equivalent to the unit interval. -/
+def stdSimplexEquivIcc : stdSimplex 𝕜 (Fin 2) ≃ Icc (0 : 𝕜) 1 where
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Basic.lean:732]: Can you add the more general
```suggestion
def stdSimplexEquivIcc : stdSimplex 𝕜 (Fin (n + 1)) ≃ stdSimplex 𝕜 (Fin n) × Icc (0 : 𝕜) 1 where
```

File: Mathlib/Analysis/Convex/Topology.lean [hunk 4]
```diff
@@ -72,6 +72,16 @@ theorem isCompact_stdSimplex : IsCompact (stdSimplex ℝ ι) :=
 instance stdSimplex.instCompactSpace_coe : CompactSpace ↥(stdSimplex ℝ ι) :=
   isCompact_iff_compactSpace.mp <| isCompact_stdSimplex _
 
+/-- The standard one-dimensional simplex in `ℝ² = Fin 2 → ℝ`
+is homeomorphic to the unit interval. -/
+def stdSimplexHomeomorphUnitInterval : stdSimplex ℝ (Fin 2) ≃ₜ unitInterval where
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Topology.lean:77]: Perhaps
```suggestion
@[simps]
def stdSimplexHomeomorphUnitInterval : stdSimplex ℝ (Fin 2) ≃ₜ unitInterval where
```

File: Mathlib/Analysis/Convex/Basic.lean [hunk 5]
```diff
@@ -688,11 +690,58 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
...
+variable {ι} [DecidableEq ι]
+
+theorem single_mem_stdSimplex (i : ι) : Pi.single i 1 ∈ stdSimplex 𝕜 ι :=
+  ⟨le_update_iff.2 ⟨zero_le_one, fun _ _ ↦ le_rfl⟩, by simp⟩
+
+theorem ite_eq_mem_stdSimplex (i : ι) : (fun j => ite (i = j) (1 : 𝕜) 0) ∈ stdSimplex 𝕜 ι := by
+  simpa only [@eq_comm _ i, ← Pi.single_apply] using single_mem_stdSimplex 𝕜 i
 #align ite_eq_mem_std_simplex ite_eq_mem_stdSimplex
 
+lemma edge_subset_stdSimplex (i j : ι) : [Pi.single i 1 -[𝕜] Pi.single j 1] ⊆ stdSimplex 𝕜 ι :=
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Basic.lean:715]:
```suggestion
lemma segment_single_subset_stdSimplex (i j : ι) : [Pi.single i 1 -[𝕜] Pi.single j 1] ⊆ stdSimplex 𝕜 ι :=
```

File: Mathlib/Analysis/Convex/Basic.lean [hunk 6]
```diff
@@ -688,11 +690,60 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
...
+
+theorem single_mem_stdSimplex (i : ι) : Pi.single i 1 ∈ stdSimplex 𝕜 ι :=
+  ⟨le_update_iff.2 ⟨zero_le_one, fun _ _ ↦ le_rfl⟩, by simp⟩
+
+theorem ite_eq_mem_stdSimplex (i : ι) : (if i = · then (1 : 𝕜) else 0) ∈ stdSimplex 𝕜 ι := by
+  simpa only [@eq_comm _ i, ← Pi.single_apply] using single_mem_stdSimplex 𝕜 i
 #align ite_eq_mem_std_simplex ite_eq_mem_stdSimplex
 
+/-- The edges are contained in the simplex. -/
+lemma segment_single_subset_stdSimplex (i j : ι) : [Pi.single i 1 -[𝕜] Pi.single j 1] ⊆ stdSimplex 𝕜 ι :=
```
Comment (urkud) [inline Mathlib/Analysis/Convex/Basic.lean:716]:
```suggestion
lemma segment_single_subset_stdSimplex (i j : ι) :
    [Pi.single i 1 -[𝕜] Pi.single j 1] ⊆ stdSimplex 𝕜 ι :=
```

File: Mathlib/Analysis/Convex/Basic.lean (hunk 3 above)
Comment (jcommelin) [reply Mathlib/Analysis/Convex/Basic.lean:735]: At least not "canonically"

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Analysis/Convex/Basic.lean (hunk 3 above)
Comment (eric-wieser) [reply Mathlib/Analysis/Convex/Basic.lean:735]: I think the true version is
```suggestion
def stdSimplexEquivIcc : stdSimplex 𝕜 (Fin 2) ≃ Σ c : Icc (0 : 𝕜) 1, c •  stdSimplex 𝕜 (Fin n) where
```
but it seems inconvenient to use

**Pull request closed without merging**